File: DailyChallenges/Scala/LeetCode_1137.scala
object Solution {
    def tribonacci(n: Int): Int = {
        val a = Array(0, 1, 1)
        if (n < 3) return a(n)
        
        for (i <- 3 to n){
            val s = a.sum; a(0) = a(1); a(1) = a(2); a(2) = s
        }
        a(2) 
    }
}

File: DailyChallenges/CPP/LeetCode_1137.cpp
class Solution {
public:
    int cache[38] = {0};
    int tribonacci(int n) {
        if (n==0) return 0;
        if (cache[n] == 0){
            if (n == 1 || n == 2) 
                cache[n] = 1;
            else
                cache[n] = tribonacci(n-1) + tribonacci(n-2) + tribonacci(n-3);
        }
        return cache[n];
    }
};
